specification/theory_operations.adoc

```diff
@@ -1,2 +1,278 @@
+| `8h`  | `SPDM_PAYLOAD_LENGTH` | 4                         | SPDM payload length.
+| `12h` | `SPDM_PAYLOAD`        | See `SPDM_PAYLOAD_LENGTH` | SPDM payload data
+                                                              (e.g. TDISP, IDE_KM).
+|===
+
+In this document, for readability reasons, flow definitions that involve SPDM
+based exchanges are simplified and do not include the above described flow
+between the TSM, the untrusted domain manager and the SPDM responder. In
+particular, the steps that cover the SPDM request generation from the TSM, the
+transmission to the host VMM, the transmission to the SPDM responder and finally
+the path back to the TSM are reduced into 2 steps:
+
+1. TSM sends SPDM request #1
+2. TSM receives SPDM response #1
+
+For example the above example would be described through those simplified steps:
+
+1. VMM calls `sbi_covh_tee_io_action()`
+2. TSM sends SPDM request #1
+3. TSM receives SPDM response #1
+4. TSM sends SPDM request #2
+5. TSM receives SPDM response #2
+6. TSM returns `sbi_covh_tee_io_action()`
+
+==== Secure SPDM Session
+
+.SPDM Session
+[source,mermaid]
+....
+%%{init: {'theme': 'neutral', 'themeVariables': {'darkMode': true}, "flowchart" : { "curve" : "basis" } } }%%
+sequenceDiagram
+
+autonumber
+
+participant VMM as Untrusted Domain Manager (VMM)
+participant TSM
+participant DSM as Device DSM or RoT
+
+VMM ->> TSM: [COVH] - sbi_covh_create_spdm_session()
+
+Note over VMM,DSM: TSM: SPDM Connection Setup
+
+TSM ->> TSM: Generate SPDM request: GET_VERSION
+TSM ->> VMM: [COVH] - spdm_req(GET_VERSION)
+VMM ->> DSM: [DOE] - SPDM_GET_VERSION
+DSM ->> VMM: [DOE] - SPDM_VERSION
+VMM ->> TSM: [COVH] - spdm_resp(VERSION)
+
+TSM ->> TSM: Generate SPDM request: GET_CAPS
+TSM ->> VMM: [COVH] - spdm_req(GET_CAPS)
+VMM ->> DSM: [DOE] - SPDM_GET_CAPS
+DSM ->> VMM: [DOE] - CAPS
+VMM ->> TSM: [COVH] - spdm_resp(CAPS)
+
+TSM ->> TSM: Generate SPDM request: NEGOTIATE_ALGS
+TSM ->> VMM: [COVH] - spdm_req(NEGOTIATE_ALGS)
+VMM ->> DSM: [DOE] - SPDM_NEGOTIATE_ALGS
+DSM ->> VMM: [DOE] - ALGS
+VMM ->> TSM: [COVH] - spdm_resp(ALGS)
+
+TSM ->> TSM: Generate SPDM request: GET_CERTIFICATE
+TSM ->> VMM: [COVH] - spdm_req(GET_CERTIFICATE)
+VMM ->> DSM: [DOE] - GET_CERTIFICATE
+DSM ->> VMM: [DOE] - CERTIFICATE
+VMM ->> TSM: [COVH] - spdm_resp(CERTIFICATE)
+TSM ->> TSM: Verify and store device certificate chain
+
+Note over VMM,DSM: TSM: SPDM Key Exchange
+
+TSM ->> TSM: Generate ephemeral SPDM keys K0
+TSM ->> TSM: Generate SPDM request: KEY_EXCHANGE(K0)
+TSM ->> VMM: [COVH] - spdm_req(KEY_EXCHANGE)
+VMM ->> DSM: [DOE] - SPDM_KEY_EXCHANGE
+DSM ->> VMM: [DOE] - KEY_RESP
+VMM ->> TSM: [COVH] - spdm_resp(KEY_RESP)
+TSM ->> TSM: Derive DHE secret
+TSM ->> TSM: Generate SPDM request: FINISH
+TSM ->> VMM: [COVH] - spdm_req(FINISH)
+VMM ->> DSM: [DOE] - SPDM_FINISH
+DSM ->> VMM: [DOE] - FINISH_RESP
+VMM ->> TSM: [COVH] - spdm_resp(FINISH_RESP)
+
+Note over VMM,DSM: TSM: SPDM Session Created
```

Comment:
When do you think we need to call SPDM GET_MEASUREMENTS?

In this API, or separate API?